### PR TITLE
Added Leadinfo Connector

### PR DESCRIPTION
**Pull request by GerbenZ01**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/Leadinfo/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,293 @@
+{
+  "swagger": "2.0",
+  "info": {
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiDefinition.swagger.json:3]: @GerbenZ01 please add the "contact" section within the "info" section. Please follow our template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json).

File: certified-connectors/Leadinfo/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    "title": "Leadinfo",
+    "description": "Leadinfo enables you to see real-time which companies are visiting your website and what pages they're viewing. By installing this app you can easily share recognized companies with your team from our portal! You need a Leadinfo account to use this App. You can request a free 14-day trial on our [website](https://portal.leadinfo.com/registration/?utm_source=microsoftteams).\n\nContact us [here](https://www.leadinfo.com/en/contact/) if you need any help.",
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiDefinition.swagger.json:5]: @GerbenZ01 is it possible to remove the special characters "\n" from the description?
I would suggest you have a proper description for the connector. and for this contact information you can have them in the contact section or in your Intro.md file. 
We will ask you to submit the Intro.md file once this PR is approved and it is time to submit to ISV Studio. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6

File: certified-connectors/Leadinfo/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+      },
+      "put": {
+        "summary": "Triggers when a company is send",
+        "description": "Triggers when a company is send",
+        "responses": {
+          "201": {
+            "description": "success"
+          }
+        },
+        "operationId": "Webhook-company-trigger",
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiDefinition.swagger.json:204]: @GerbenZ01 all operationIds should be in Camel Case. Please change to "WebhookCompanyTrigger". Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations).
Please fix all operationIds in the swagger file.

File: certified-connectors/Leadinfo/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    }
+  },
+  "security": [
+    {
+      "oauth2-auth": [
+        "webhook"
+      ]
+    }
+  ],
+  "tags": []
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiDefinition.swagger.json:292]: @GerbenZ01 please add the "x-ms-connector-metadata" property to the swagger file. Please see our template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiDefinition.swagger.json#:~:text=%22x%2Dms%2Dconnector,%5D).

File: certified-connectors/Leadinfo/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            "refreshUrl": {
+              "value": "https://api.leadinfo.com/oauth2/token"
+            }
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#FA2E48",
+    "capabilities": [],
+    "publisher": "Leadinfo"
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiProperties.json:34]: @GerbenZ01 Please add the "stackOwner" property. please see our template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/apiProperties.json).
```
"stackOwner": "Name of the owner of the service or product."
```

File: certified-connectors/Leadinfo/readme.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# Leadinfo Certified Connector
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/readme.md:1]: @GerbenZ01 Please follow and include all required section in the readme file. Please follow our template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md), and for raw code view [here](https://raw.githubusercontent.com/microsoft/PowerPlatformConnectors/dev/templates/certified-connectors/readme.md).

File: certified-connectors/Leadinfo/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+              ]
+            }
+          }
+        ]
+      }
+    },
+    "/v1/webhook/{id}": {
+      "delete": {
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiDefinition.swagger.json:252]: @GerbenZ01 it is not allowed to have a default response only. is it possible to define a proper response. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/create-webhook-trigger#creating-the-webhook).

File: certified-connectors/Leadinfo/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+            "description": "default",
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiDefinition.swagger.json:253]: @GerbenZ01 it is possible to change this to a more proper description. something like "OK", "Success", "Bad Request"..etc.

File: certified-connectors/Leadinfo/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiDefinition.swagger.json:254]: @GerbenZ01 empty schema is not allowed. please define a proper schema.

File: certified-connectors/Leadinfo/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+            "branch_code_international": {
+              "type": "string",
+              "description": "branch_code_international"
+            }
+          }
+        },
+        "description": "webhook response"
+      },
+      "put": {
+        "summary": "Triggers when a company is send",
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiDefinition.swagger.json:197]: @GerbenZ01 all files should be free of grammatical errors and typos. I think here meant to say "Triggers when a company is sent"

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by Amjed-Ayoub** [comment]
@GerbenZ01 May I ask you please to sign our CLA. you can do so by commenting with the below command.
```
@microsoft-github-policy-service agree company="your company"
```

**Comment by Amjed-Ayoub** [comment]
Hello @GerbenZ01 ,
I hope you are doing well.
I am following up on my messages to see if you need any help with updating your connector artifacts.
Please do not hesitate to reach out to me if you have any questions as we are always here to help. 😊
I look forward to hearing from you.
Thank you very much.

**Comment by Amjed-Ayoub** [comment]
Hello @GerbenZ01 ,

I am sending this email to follow up with you to see how things are going for getting your connector certified 😊.

Please let me know if you have any questions.

I look forward to hearing from you soon.

Thank you so much for working with us.

**Comment by GerbenZ01** [comment]
Hello Amjed,

Sorry for my late response, but I was really busy with some other things
that were prioritised higher. Therefore I didn't have any time to look into
the errors up until now. I pushed another commit to the repo with the
fixes, and here is the signed Contributor License Agreement:

@microsoft-github-policy-service agree company=Leadinfo

Do I need to add the signed Contributor License Agreement anywhere else?

With kind regards,


Gerben Zwemer

Developer

Leadinfo



***@***.***
www.leadinfo.com <//www.leadinfo.com>
Rivium Quadrant 141







Op di 6 jun 2023 om 01:14 schreef Amjed Ayoub ***@***.***>:

File: certified-connectors/Leadinfo/apiDefinition.swagger.json [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+      "propertyName": "Website",
+      "propertyValue": "https://www.leadinfo.com/en/"
+    },
+    {
+      "propertyName": "Privacy policy",
+      "propertyValue": "https://www.leadinfo.com/en/privacy/"
+    },
+    {
+      "propertyName": "Categories",
+      "propertyValue": "https://www.leadinfo.com/en/eula/"
```
Comment (Amjed-Ayoub) [inline certified-connectors/Leadinfo/apiDefinition.swagger.json]: @GerbenZ01 The connector category is invalid. Available categories are: [AI, Business Management, Business Intelligence, Collaboration, Commerce, Communication, Content and Files, Finance, Data, Human Resources, Internet of Things, IT Operations, Lifestyle and Entertainment, Marketing, Productivity, Sales and CRM, Security, Social Media, Website].
You can select up to two categories and please separate them with a semicolon.

**Comment by Amjed-Ayoub** [comment]
> @microsoft-github-policy-service agree company=Leadinfo

@GerbenZ01 please make a new comment with this command only.

**Comment by GerbenZ01** [comment]
@microsoft-github-policy-service agree company=Leadinfo

**Comment by GerbenZ01** [comment]
Hey Amjed,

Just made a new comment in the pull request.

With kind regards,

Gerben Zwemer

Developer

Leadinfo



***@***.***
www.leadinfo.com <//www.leadinfo.com>
Rivium Quadrant 141







Op wo 14 jun 2023 om 01:35 schreef Amjed Ayoub ***@***.***>:

File: certified-connectors/Leadinfo/apiDefinition.swagger.json (hunk 11 above)
Comment (Amjed-Ayoub) [reply certified-connectors/Leadinfo/apiDefinition.swagger.json]: @GerbenZ01 please select  a category from the list above. please change "https://www.leadinfo.com/en/eula/" as it is not a valid value.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by GerbenZ01** [comment]
Hey Amjed,

I changed the category.

With kind regards,

Gerben Zwemer

Developer

Leadinfo



***@***.***
www.leadinfo.com <//www.leadinfo.com>
Rivium Quadrant 141







Op ma 19 jun 2023 om 20:31 schreef Amjed Ayoub ***@***.***>:

**Review by Amjed-Ayoub — APPROVED** [review]
Hello @GerbenZ01 ,
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector. 
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.

**Pull request merged**